ckanext/saml2/plugin.py

```diff
@@ -176,6 +179,46 @@ def get_came_from(relay_state):
     return came_from.encode('utf8')
 
 
+def saml2_get_user_gen(id=None, return_gen=False):
+    query = model.Session.query(UserSsoGen).filter(or_(
+                                    UserSsoGen.user_name == id,
+                                    UserSsoGen.gen == id,
+                                    UserSsoGen.id == id))
+    if return_gen:
+        query = query.first()
+        return query if query is None else query.gen
+    else:
+        return query
+
+
+def saml_change_user_state_from_saml2_sso_gen_db(query, state):
+    if query is not None:
+        if state == 'deleted':
+            query.update({'state': 'deleted'})
+        elif state == 'active':
+            query.update({'state': 'active'})
+        model.Session.commit()
+
+
+def saml2_user_delete(context, data_dict):
```

Comment:
The param handling needs some work. Since it is overriding the `user_delete` action, it should continue to handle the `id` argument in the standard way, ie., as a user name or id.  Name ID (GEN) handling should be an additional feature, ie., only look for a NameID in data_dict if id is not present.